src/index.js

```diff
@@ -20,19 +36,121 @@ const displayDataTable = (image, title) => {
+  target.parentElement.parentElement.remove();
+};
+
+const countItems = () => {
+  const itemCount = document.querySelector('#item-count');
+  const mySection = document.querySelector('body > main > section');
+  itemCount.firstChild.innerHTML = `Exhibition ${countElements(mySection)}`;
+};
+
+const countComment = () => {
+  const commentCount = document.getElementById('comment-container');
+  commentCount.previousElementSibling.innerHTML = `Comments ${countElements(commentCount)}`;
+};
+
+function showComment(user, str) {
+  const ulContainer = document.querySelector('#comment-container');
+  const li = document.createElement('li');
+  li.innerHTML = `${user} : ${str}`;
+  ulContainer.appendChild(li);
+}
+
+function displayComments(id) {
+  const showProper = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/9mAPgvMc6PjOJk4JU1ZU/comments?item_id=${id}`;
+  getData(showProper)
+    .then((data) => data.forEach((elem) => showComment(elem.username, elem.comment)))
+    .then(() => countComment());
+}
+
+function addComment(id, user, str) {
+  const data = {
+    item_id: id,
+    username: user,
+    comment: str,
+  };
+
+  postData(commentLink, data)
     .then((data) => {
-      displayData.innerHTML = '';
-      data.forEach((element) => displayDataTable(element.hdurl, element.title));
+      if (data.status === 201) {
+        showComment(user, str);
+      }
     });
+}
+
+const displayImagePopup = (id) => {
+  getData(generateImage(id))
+    .then((data) => commentPopup(data.hdurl, data.title, data.explantion, id))
+    .then(() => {
+      displayComments(id);
+      const closeBtn = document.getElementById('close-btn');
+      closeBtn.addEventListener('click', () => {
+        closeDisplayPop(closeBtn);
+      });
+    });
+};
+
+const clickLove = (id, love) => {
+  const clickLike = document.getElementById(id);
+  clickLike.parentElement.nextElementSibling.innerHTML = `${love} likes`;
 };
 
+function displayLike() {
+  getData(starLink)
+    .then((data) => data.forEach((elem) => clickLove(elem.item_id, elem.likes)));
+}
+
+function displayFromApi() {
+  getData(generateUrl)
+    .then((data) => data.forEach((elem, index) => displayDataTable(elem.hdurl, elem.title, index)))
+    .then(() => {
+      displayLike();
+      countItems();
+    });
+}
+
+function giveLikes(id, stars) {
+  const data = { item_id: id };
+  postData(starLink, data)
+    .then((data) => {
+      if (data.status === 201) {
+        clickLove(id, stars);
+      }
+    });
+}
+
 displayFromApi();
+
+displayData.addEventListener('click', (e) => {
+  if (e.target.classList.contains('fa-heart')) {
+    e.preventDefault();
+    const oneL = parseInt(e.target.parentElement.parentElement.nextElementSibling.textContent, 10);
+    const likes = oneL + 1;
+    giveLikes(e.target.parentElement.id, likes);
+  }
+  if (e.target.classList.contains('fa-comment')) {
+    displayImagePopup(parseInt(e.target.id, 10));
+  }
```

Comment:
- [ ] Please, it would be better to fix the id that `displayImagePopup` is receiving because the `e.target.id` doesn't exit. (e.target is the icon) 🔨 For this reason, your popup doesn't show the comments.

![id-comment](https://user-images.githubusercontent.com/79876959/144726004-a6444dac-428d-467d-b8f9-a302ffac2122.PNG)